cachi2/core/models/sbom.py

```diff
@@ -93,3 +98,207 @@ class Sbom(pydantic.BaseModel):
     def _unique_components(cls, components: list[Component]) -> list[Component]:
         """Sort and de-duplicate components."""
         return unique_sorted(components, by=lambda component: component.key())
+
+    def to_spdx(self) -> "SPDXSbom":
+        """Convert a CycloneDX SBOM to an SPDX SBOM."""
+        packages = []
+        for component in self.components:
+            annotations = []
+            for prop in component.properties:
+                annotations.append(
+                    SPDXPackageAnnotation(
+                        annotator="cachi2",
+                        annotationDate=datetime.datetime.now().isoformat(),
+                        annotationType="OTHER",
+                        comment=json.dumps({"name": f"{prop.name}", "value": f"{prop.value}"}),
+                    )
+                )
+            packages.append(
+                SPDXPackage(
+                    name=component.name,
+                    versionInfo=component.version,
+                    externalRefs=[
+                        SPDXPackageExternalRef(
+                            referenceCategory="PACKAGE-MANAGER",
+                            referenceLocator=component.purl,
+                            referenceType="purl",
+                        )
+                    ],
+                    annotations=annotations,
+                )
+            )
+        return SPDXSbom(
+            packages=packages,
+            creationInfo=SPDXCreationInfo(
+                creators=[f"{tool.vendor} {tool.name}" for tool in self.metadata.tools]
+            ),
+        )
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: str = pydantic.Field(
+        pattern="^SECURITY|PACKAGE-MANAGER|PERSISTENT-ID|OTHER$"
```

Comment:
you mean str instead of Litral? I haven't really used Literal type before so I'm used to put everywhere str by default. But yeah, Literal could be used here.